collections/src/main/scala/strawman/collection/ArrayOps.scala

```diff
@@ -63,6 +68,8 @@ class ArrayOps[A](val xs: Array[A]) extends AnyVal
 
   def zip[B: ClassTag](that: Iterable[B]): Array[(A, B)] = fromTaggedIterable(View.Zip(toIterable, that))
 
+  def zipWithIndex(implicit ct: ClassTag[(A, Int)]): Array[(A, Int)] = fromTaggedIterable(View.ZipWithIndex(toIterable))
```

Comment:
How did you manage to solve the ambiguity with the inherited version?

Reply:
I misunderstood the error message, the ambiguity was on `WithFilter.map`, which has two overloads. I moved one in a parent trait in this PR.